### PR TITLE
remove major LF dev version

### PR DESCRIPTION
**Pull request by S11001001**
Part of #179; removes LF dev "major" version while leaving some scaffolding in place that will be needed for the dev "minor" version support.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by hurryabit** [comment]
This looks like the dev version does not contain the hash anymore. I can live with this but want to make sure it is intended.

**Comment by S11001001** [comment]
@hurryabit See discussion on #179; intent is to remove, but we'll see if something pops up in implementation.

File: daml-foundations/daml-ghc/package-database/BUILD.bazel [hunk 1]
```diff
@@ -60,15 +60,7 @@ daml_package_rule(
...
-daml_package_rule(
-    name = "daml-prim-dev",
-    srcs = "//daml-foundations/daml-ghc/daml-prim-src",
-    daml_lf_version = "dev",
-    package_db = ":package_db_for_daml-prim",
-    pkg_name = "daml-prim",
-    pkg_root = "daml-foundations/daml-ghc/daml-prim-src",
-    visibility = ["//visibility:public"],
-)
+# TODO SC: revert a21570717b and update for 1.dev https://github.com/digital-asset/daml/issues/179
```
Comment (S11001001) [inline daml-foundations/daml-ghc/package-database/BUILD.bazel]: https://github.com/digital-asset/daml/pull/681/commits/a21570717b4541e16a171c3b67887b3a3befb941

File: daml-lf/archive/da/daml_lf.proto [hunk 2]
```diff
@@ -19,7 +18,7 @@ message ArchivePayload {
     daml_lf_0.Package daml_lf_0 = 1;
     daml_lf_1.Package daml_lf_1 = 2;
 
-    daml_lf_dev.Package daml_lf_dev = 9999;
```
Comment (bitonic) [inline daml-lf/archive/da/daml_lf.proto]: please reserve `9999`, and add a comment

Comment (S11001001) [reply daml-lf/archive/da/daml_lf.proto]: 9cadc34e75

Comment (S11001001) [reply daml-lf/archive/da/daml_lf.proto]: bf29e0e97910ac043ca13406cabf0259f8e98aca

**Pull request merged**